Assets/Player/CameraControls.gd

```diff
@@ -0,0 +1,103 @@
+extends Node
+
+const ZOOM_IN_LIMIT = 5
+const ZOOM_OUT_LIMIT = 35
+const ZOOM_VALUE = 5
+const RAY_LENGTH = 1000
+
+const MOVE_SPEED = 10
+const MOVE_FASTER_MULT = 3
+
+export(NodePath) var origin_path: NodePath
+export(NodePath) var camera_path: NodePath
+export(NodePath) var rotation_y_path: NodePath
+
+onready var _origin := get_node(origin_path) as Spatial
+onready var _camera := get_node(camera_path) as Camera
+onready var _rotation_y := get_node(rotation_y_path) as Spatial
+onready var _viewport := get_viewport() as Viewport
+onready var _viewport_size := _viewport.size as Vector2
+onready var _viewport_aspect := _viewport_size.aspect() as float
+
+var _basis: Basis
+var _move_drag_start: Vector2
+var enabled: bool = true setget set_enabled, get_enabled
+
+func _ready():
```

Comment:
Add parameter types and return types to all functions. If there have been existing functions without them, it was most likely an oversight.

`func _ready() -> void:`